lib/hotel_web/schema/middleware/changeset_errors.ex

```diff
@@ -0,0 +1,33 @@
+defmodule HotelWeb.Schema.Middleware.ChangesetErrors do
+  @moduledoc """
+  This module contains helper functions for formatting changeset errors into a
+  nice response for our resolvers
+  """
+  @behaviour Absinthe.Middleware
+
+  @doc """
+  This function takes on Absinthe resolution and checks it for Changeset Errors
+  if it finds errors it formats them into a helpful fashion
+  """
+
+  @spec call(map(), any()) :: map()
+    def call(res, _) do
+      %{res | errors: Enum.flat_map(res.errors, &transform_error/1)}
+  end
+    @spec transform_error(Ecto.Changeset.t()) :: String.t()
+    defp transform_error(%Ecto.Changeset{} = changeset) do
+      changeset
+        |> Ecto.Changeset.traverse_errors(&build_error/1)
+        |> Enum.map(fn {item, error} ->"#{item}: #{error}" end)
+  end
+    #handle other kind of errors
+    defp transform_error(error), do: [error]
+
+    #build error using options
+    @spec build_error(tuple()) :: String.t()
+    defp build_error({msg, opts}) do
+      Enum.reduce(opts, msg, fn({key, value}, acc) ->
+        String.replace(acc, "%{#{key}}", to_string(value))
```

Comment:
@WanjikuMac  this error could include more details usually in an `opts` field

Reply:
Could you please elaborate or do you mean something like this?
```defp build_error({msg, opts(attrs)}) do```

Reply:
I mean the `opts` item has more details usually about the error that the user can use to recover

Reply:
@zacck Oh okay, how would I then can I added a detailed error msg with even a couple of solutions?

Reply:
like this https://github.com/LittleKidogo/quick_bytes_backend/blob/3f8c1d75b26c5835f70eed3a464351e806a104d9/lib/qb_backend_web/schema/middleware/changeset_errors.ex#L30

Reply:
@zacck Which is the same thing I have ama?